pypeit/spectrographs/p200_dbsp.py

```diff
@@ -0,0 +1,422 @@
+        meta['idname'] = dict(ext=0, card='IMGTYPE')
+        # Lamps
+        meta['lampstat01'] = dict(ext=0, card='LAMPS')
+
+        # Ingest
+        self.meta = meta
+
+    def compound_meta(self, headarr: List[fits.Header], meta_key: str):
+        """
+        Methods to generate meta in a more complex manner than simply
+        reading from the header.
+
+        mjd is converted from UTSHUT header
+        dispangle is parsed from ANGLE header
+
+        Args:
+            headarr: List[fits.Header]
+              List of headers
+            meta_key: str
+
+        Returns:
+            value:
+
+        """
+        if meta_key == 'mjd':
+            return Time(headarr[0]['UTSHUT']).mjd
+        elif meta_key == 'dispangle':
+            try:
+                return Angle(headarr[0]['ANGLE']).deg
+            except Exception as e:
+                print(headarr[0]['ANGLE'])
+                raise e
+        else:
+            return None
+
+    def pypeit_file_keys(self):
+        pypeit_keys = super(P200DBSPSpectrograph, self).pypeit_file_keys()
+        pypeit_keys += ['slitwid']
+        return pypeit_keys
+    
+    def check_frame_type(self, ftype, fitstbl, exprng=None):
+        """
+        Check for frames of the provided type.
+        """
+        good_exp = framematch.check_frame_exptime(fitstbl['exptime'], exprng)
+        if ftype in ['science', 'standard']:
+            return good_exp & (fitstbl['lampstat01'] == '0000000') & (fitstbl['idname'] == 'object')
+        if ftype == 'bias':
+            return good_exp & (fitstbl['idname'] == 'bias')
+        if ftype in ['pixelflat', 'trace', 'illumflat']:
+            return good_exp & (fitstbl['idname'] == 'flat')
+        if ftype in ['pinhole', 'dark']:
+            # Don't type pinhole or dark frames
+            return np.zeros(len(fitstbl), dtype=bool)
+        if ftype in ['arc', 'tilt']:
+            return good_exp & (fitstbl['lampstat01'] != '0000000') & (fitstbl['idname'] == 'cal')
+        msgs.warn('Cannot determine if frames are of type {0}.'.format(ftype))
+        return np.zeros(len(fitstbl), dtype=bool)
+
+
+class P200DBSPBlueSpectrograph(P200DBSPSpectrograph):
+    """
+    Child to handle P200/DBSP blue specific code
+    """
+    def __init__(self):
+        # Get it started
+        super(P200DBSPBlueSpectrograph, self).__init__()
+        self.spectrograph = 'p200_dbsp_blue'
+        self.camera = 'DBSPb'
+    
+    def compound_meta(self, headarr: List[fits.Header], meta_key: str):
+        """
+        Methods to generate meta in a more complex manner than simply
+        reading from the header. Super method handles mjd and dispangle
+
+        binning is parsed from CCDSUM header
+
+        Args:
+            headarr: List[fits.Header]
+              List of headers
+            meta_key: str
+
+        Returns:
+            value:
+
+        """
+        retval = super(P200DBSPBlueSpectrograph, self).compound_meta(headarr, meta_key)
```

Comment:
I'm sure this is right, but I'm not following why this line is needed here - maybe add an inline comment?